### PR TITLE
Rename quarks resources

### PR DESCRIPTION
**Pull request by edwardstudy**
[#169095334](https://www.pivotaltracker.com/story/show/169095334)

Requires: cloudfoundry-incubator/quarks-job#20

### CODE REVIEW

**Comment by edwardstudy** [comment]
It should be running now.

**Comment by edwardstudy** [comment]
Need to merge #701 firstly.

**Comment by qu1queee** [comment]
@edwardstudy the test are failing in many specs, is there any other dependency that this PR needs?

File: README.md [hunk 1]
```diff
@@ -62,10 +62,10 @@ webhooks:
 
 The `cf-operator` watches four different types of custom resources:
 
-- [BoshDeployment](docs/controllers/boshdeployment.md)
-- [ExtendedJob](https://github.com/cloudfoundry-incubator/quarks-job/blob/master/docs/extendedjob.md)
-- [ExtendedSecret](docs/controllers/extendedsecret.md)
-- [ExtendedStatefulset](docs/controllers/extendedstatefulset.md)
+- [BoshDeployment](docs/controllers/bosh_deployment.md)
+- [QuarksJob](https://github.com/cloudfoundry-incubator/quarks-job/blob/master/docs/quarksjob.md)
```
Comment (rohitsakala) [inline README.md]: Shouldn't this be `quarks_job.md`

File: cmd/internal/root.go [hunk 2]
```diff
@@ -148,9 +147,8 @@ func init() {
 	pf.StringP("bosh-dns-docker-image", "", "coredns/coredns:1.6.3", "The docker image used for emulating bosh DNS (a CoreDNS image)")
 	pf.String("cluster-domain", "cluster.local", "The Kubernetes cluster domain")
 	pf.Int("max-boshdeployment-workers", 1, "Maximum number of workers concurrently running BOSHDeployment controller")
-	pf.Int("max-extendedjob-workers", 1, "Maximum number of workers concurrently running ExtendedJob controller")
-	pf.Int("max-extendedsecret-workers", 5, "Maximum number of workers concurrently running ExtendedSecret controller")
-	pf.Int("max-extendedstatefulset-workers", 1, "Maximum number of workers concurrently running ExtendedStatefulSet controller")
+	pf.Int("max-quarks-secret-workers", 5, "Maximum number of workers concurrently running QuarksSecret controller")
```
Comment (rohitsakala) [inline cmd/internal/root.go]: Is `quarkssecret` not a single word here ?

File: cmd/internal/root.go [hunk 3]
```diff
@@ -159,9 +157,8 @@ func init() {
 		"bosh-dns-docker-image",
 		"cluster-domain",
 		"max-boshdeployment-workers",
-		"max-extendedjob-workers",
-		"max-extendedsecret-workers",
-		"max-extendedstatefulset-workers",
+		"max-quarks-secret-workers",
+		"max-quarks-statefulset-workers",
```
Comment (rohitsakala) [inline cmd/internal/root.go]: Is `quarkssecret` not a single word here ?

File: cmd/internal/root.go [hunk 4]
```diff
@@ -171,9 +168,8 @@ func init() {
 	argToEnv["bosh-dns-docker-image"] = "BOSH_DNS_DOCKER_IMAGE"
 	argToEnv["cluster-domain"] = "CLUSTER_DOMAIN"
 	argToEnv["max-boshdeployment-workers"] = "MAX_BOSHDEPLOYMENT_WORKERS"
-	argToEnv["max-extendedjob-workers"] = "MAX_EXTENDEDJOB_WORKERS"
-	argToEnv["max-extendedsecret-workers"] = "MAX_EXTENDEDSECRET_WORKERS"
-	argToEnv["max-extendedstatefulset-workers"] = "MAX_EXTENDEDSTATEFULSET_WORKERS"
+	argToEnv["max-quarks-secret-workers"] = "MAX_QUARKS_SECRET_WORKERS"
```
Comment (rohitsakala) [inline cmd/internal/root.go]: Is `quarkssecret` not a single word here ?

File: docs/controllers/quarks_statefulset.md [hunk 5]
```diff
@@ -17,35 +17,35 @@
          11. [Volume Management](#volume-management)
          12. [AZ Support](#az-support)
          13. [Active/passive model](#activepassive-model)
-      2. [Statefulset Cleanup Controller](#statefulset-cleanup-controller)
+      2. [Statefulset Cleanup Controller](#_statefulset-cleanup-controller_)
```
Comment (rohitsakala) [inline docs/controllers/quarks_statefulset.md]: The underscrores should be removed.

File: docs/from_bosh_to_kube.md [hunk 6]
```diff
@@ -14,8 +14,8 @@
     - [Misc](#Misc)
   - [Conversion Details](#Conversion-Details)
     - [Calculation of docker image location for releases](#Calculation-of-docker-image-location-for-releases)
-    - [Variables to Extended Secrets](#Variables-to-Extended-Secrets)
-    - [Instance Groups to Extended StatefulSets and Jobs](#Instance-Groups-to-Extended-StatefulSets-and-Jobs)
+    - [Variables to QuarksSecret Secrets](#Variables-to-Quarks-Secrets)
```
Comment (rohitsakala) [inline docs/from_bosh_to_kube.md]: Secrets is repeated twice.

File: docs/from_bosh_to_kube.md [hunk 7]
```diff
@@ -344,14 +344,14 @@ addons:
...
-# For each variable, the cf-operator creates ExtendedSecrets
+# For each variable, the cf-operator creates QuarksSecrets
 # As with normal BOSH, variables are referenced by job properties.
 # Each variable's generated secret is mounted in the container that renders each job's
 # templates. They are then used by the rendering process.
 # This means that the operator needs to look at the job's properties, and parse any references
 # to variables, so it knows what it needs to mount.
 variables:
-    # Unique name used to identify a variable. Used to name the ExtendedSecret
+    # Unique name used to identify a variable. Used to name the EQuarksSecret
```
Comment (rohitsakala) [inline docs/from_bosh_to_kube.md]: I think `E` should be removed.

File: docs/from_bosh_to_kube.md [hunk 8]
```diff
@@ -507,7 +507,7 @@ Providing credentials for private registries is supported by Kubernetes. Please
 ### Running manual errands
 
 BOSH makes use of errands, which are manually triggered.
-We support manual triggers - you can learn more in the [ExtendedJob docs](controllers/extendedjob.md#errand-jobs).
+We support manual triggers - you can learn more in the [quarksJob docs](https://github.com/cloudfoundry-incubator/quarks-job/blob/master/docs/quarksjob.md#errands).
```
Comment (rohitsakala) [inline docs/from_bosh_to_kube.md]: Capital `Q`

File: e2e/cli/cli_test.go [hunk 9]
```diff
@@ -41,9 +41,8 @@ var _ = Describe("CLI", func() {
   -c, --kubeconfig string                        \(KUBECONFIG\) Path to a kubeconfig, not required in-cluster
   -l, --log-level string                         \(LOG_LEVEL\) Only print log messages from this level onward \(default "debug"\)
       --max-boshdeployment-workers int           \(MAX_BOSHDEPLOYMENT_WORKERS\) Maximum number of workers concurrently running BOSHDeployment controller \(default 1\)
-      --max-extendedjob-workers int              \(MAX_EXTENDEDJOB_WORKERS\) Maximum number of workers concurrently running ExtendedJob controller \(default 1\)
-      --max-extendedsecret-workers int           \(MAX_EXTENDEDSECRET_WORKERS\) Maximum number of workers concurrently running ExtendedSecret controller \(default 5\)
-      --max-extendedstatefulset-workers int      \(MAX_EXTENDEDSTATEFULSET_WORKERS\) Maximum number of workers concurrently running ExtendedStatefulSet controller \(default 1\)
+      --max-quarks-secret-workers int            \(MAX_QUARKS_SECRET_WORKERS\) Maximum number of workers concurrently running QuarksSecret controller \(default 5\)
```
Comment (rohitsakala) [inline e2e/cli/cli_test.go]: is `quarkssecret` not a single word ?

File: e2e/kube/examples_test.go [hunk 10]
```diff
@@ -32,31 +32,31 @@ var _ = Describe("Examples Directory", func() {
...
-			example = "extended-statefulset/exstatefulset_configs.yaml"
+			example = "quarks-statefulset/qstatefulset_configs.yaml"
 		})
 
 		It("creates and updates statefulsets", func() {
 			By("Checking for pods")
-			podWait("pod/example-extendedstatefulset-v1-0")
-			podWait("pod/example-extendedstatefulset-v1-1")
+			podWait("pod/example-quarks-statefulset-v1-0")
+			podWait("pod/example-quarks-statefulset-v1-1")
```
Comment (rohitsakala) [inline e2e/kube/examples_test.go]: is `quarksstatefulset` not a single word ?

File: pkg/bosh/manifest/bpm_secret_info.go [hunk 11]
```diff
@@ -5,17 +5,17 @@ import (
...
+// instance group which matters for quarksStatefulSet pods
 // such as AZ's, instance group count and BPM Configs
 type BPMInfo struct {
 	InstanceGroup BPMInstanceGroup `json:"instance_group,omitempty"`
 	Configs       bpm.Configs      `json:"configs,omitempty"`
 	Variables     []Variable       `json:"variables,omitempty"`
 }
 
-// BPMInstanceGroup is a custome instance group spec
+// BPMInstanceGroup is a costume instance group spec
```
Comment (rohitsakala) [inline pkg/bosh/manifest/bpm_secret_info.go]: `custom`

File: pkg/kube/controllers/boshdeployment/bpm_reconciler.go [hunk 12]
```diff
@@ -196,21 +195,21 @@ func (r *ReconcileBPM) applyBPMResources(bpmSecret *corev1.Secret, manifest *bdm
 		return nil, errors.Errorf("instance group '%s' not found", instanceGroupName)
 	}
 
-	// Fetch exsts version
-	exstendedStatefulSet := &essv1.ExtendedStatefulSet{}
-	exstendedStatefulSetName := instanceGroup.ExtendedStatefulsetName(manifest.Name)
-	err := r.client.Get(r.ctx, types.NamespacedName{Namespace: r.config.Namespace, Name: exstendedStatefulSetName}, exstendedStatefulSet)
+	// Fetch qstscontroller version
```
Comment (rohitsakala) [inline pkg/kube/controllers/boshdeployment/bpm_reconciler.go]: its `qsts` not controller.

File: cmd/internal/root.go (hunk 2 above)
Comment (edwardstudy) [reply cmd/internal/root.go]: Same as quarks-job, and quarks-statefulset.

File: README.md (hunk 1 above)
Comment (edwardstudy) [reply README.md]: It belongs to the quarks-job scope.

File: cmd/internal/root.go (hunk 2 above)
Comment (rohitsakala) [reply cmd/internal/root.go]: Okies.

File: README.md (hunk 1 above)
Comment (rohitsakala) [reply README.md]: Okies.

**Pull request merged**